openedx_tagging/core/tagging/import_export/api.py

```diff
@@ -0,0 +1,176 @@
+"""
+Import/export API functions
+
+A modular implementation has been followed for both functionalities.
+
+Import
+------------
+
+In this functionality we have the following pipeline with the following classes:
+
+Parser.parse_import() -> TagImportPlan.generate_actions() -> [ImportActions]
+-> TagImportPlan.plan() -> TagImportPlan.execute()
+
+Parsers are in charge of reading the input file,
+making the respective verifications of its format and returning a list of TagItems.
+You need to create parser for each format that the system will accept.
+For more information see parsers.py
+
+TagImportPlan receives a list of TagItems. With this, it generates each
+action that will be executed in the import.
+Each Action are in charge of verifying and executing specific
+and simple operations in the database, such as creating or rename tag.
+For more information see actions.py
+
+In each action it is verified if there are no errors or inconsistencies
+with taxonomy tags or with previous actions.
+In the end, TagImportPlan contains all actions and possible errors.
+You can run `plan()` to see the actions and errors or you can run `execute()`
+to execute each action.
+
+Export
+----------
+
+The export only uses Parsers. Calls the respective function and
+returns a string with the data.
+"""
+from io import BytesIO
+
+from django.utils.translation import gettext_lazy as _
+
+from ..models import Taxonomy, TagImportTask, TagImportTaskState
+from .parsers import get_parser, ParserFormat
+from .import_plan import TagImportPlan, TagImportTask
+
+
+def import_tags(
+    taxonomy: Taxonomy,
+    file: BytesIO,
+    parser_format: ParserFormat,
+    replace=False,
+) -> bool:
+    """
+    Execute the necessary actions to import the tags from `file`
+
+    You can read the docstring of the top for more info about the
+    modular architecture.
+
+    It creates an TagImporTask to keep logs of the execution
+    of each import step and the current status.
+    There can only be one task in progress at a time per taxonomy
+    """
+
+    _import_export_validations(taxonomy)
+
+    # Checks that exists only one tag
+    if not _check_unique_import_task(taxonomy):
+        raise ValueError(
```

Comment:
In the future, we may need a function to force clean the status of an import task to avoid a lock. Maybe we end up with an invalid state (a task not in SUCCESS nor ERROR) caused by a server crash or an unforeseen condition that prevents us from doing further imports.

I don't think this is a problem for the MVP.

Reply:
Good catch! I added a list of TODOs on the [docstring with this case]( https://github.com/openedx/openedx-learning/pull/64/commits/1577938e8ac1c6af7e3714fd6f75db71cda9f3c2#diff-5ba76c1e98a2c00334984b6209fa173366c6996a16478f352d21a8f62b9c8189R38)